pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerBuilderImpl.java

```diff
@@ -213,18 +214,21 @@ private ProducerBuilderImpl(PulsarClientImpl client, ProducerConfigurationData c
 
     @Override
     public ProducerBuilder<T> batchingMaxPublishDelay(long batchDelay, @NonNull TimeUnit timeUnit) {
+        checkArgument(batchDelay >= 0, "batchDelay needs to be >= 0");
```

Comment:
Why this can’t be negative?

Reply:
Firstly, thanks all for the review.

There is the similar check _deprecated_ `ProducerConfiguration.setBatchingMaxPublishDelay` and could not see any use-case for negative access on code-base (**e.g:** Unit Test case)
In general, i was thinking that this kind of validation can be useful due to _public API_. However, if there is valid use-case, it can be skipped and UT can be useful to highlight negative case.

Reply:
if ProducerConfiguration already verifies it, why do we need to the verification in builder? Also `ProducerConfiguration.setBatchingMaxPublishDelay` uses `>= 1` as the verification rule, but you are using `>=0`. what is the thought behind this?

Reply:
Thanks for review and sorry for the late reply.

`ProducerConfigurationData` is being wrapped by `ProducerBuilderImpl` and `ProducerConfiguration`. However, i have mentioned `ProducerConfiguration`, not `ProducerConfigurationData`.

So, my point is that these validations exist on **deprecated** `ProducerConfiguration` API and not exist **new** `ProducerBuilderImpl` API. Does it make sense to align both of them as latest submitted patch?

Reply:
If ProducerConfigurationData already does the verification, I don't think we need to redo the verification at ProducerBuilderImpl. We should do the verification at the common class which already exists in ProducerConfigurationData. so it might be making more sense to remove the verification from deprecated ProducerConfiguration if ProducerConfigurationData already did it.

Reply:
I agree to keep these kind of validation in single tier and support required APIs by avoiding code duplication. 

So, currently, the following `setters` look like requiring alignment on `ProducerConfigurationData`.
```
setProducerName
setSendTimeoutMs
setMaxPendingMessages
setMaxPendingMessagesAcrossPartitions
setBatchingMaxPublishDelayMicros
setBatchingMaxMessages
```
Will be submitting new patch if we do not have any concern.

Reply:
sgtm